ipn/ipnlocal/profiles.go

```diff
@@ -0,0 +1,435 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipnlocal
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+	"time"
+
+	"tailscale.com/ipn"
+	"tailscale.com/types/logger"
+	"tailscale.com/util/strs"
+)
+
+type loginProfile struct {
+	ID   string
+	Name string
```

Comment:
Can we persist a few more fields from the login? Ideally `DisplayName`, `LoginName`, and `ProfilePicURL`, so that we have have enough information to render the Mac menu item for all profiles?